README.md
[![Latest Stable Version](https://img.shields.io/packagist/v/crazy-max/cws-curl.svg?style=flat-square)](https://packagist.org/packages/crazy-max/cws-curl)
[![Minimum PHP Version](https://img.shields.io/badge/php-%3E%3D%205.3.0-8892BF.svg?style=flat-square)](https://php.net/)
[![Build Status](https://img.shields.io/travis/com/crazy-max/CwsCurl/master.svg?style=flat-square)](https://travis-ci.com/crazy-max/CwsCurl)
[![Code Quality](https://img.shields.io/codacy/grade/bf248165410141b0a323d9ac9fe49248.svg?style=flat-square)](https://www.codacy.com/app/crazy-max/CwsCurl)
[![Donate Paypal](https://img.shields.io/badge/donate-paypal-7057ff.svg?style=flat-square)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=ECNZJ3UYAX7PU)

## About

A flexible wrapper PHP class for the cURL extension.

## Installation

```bash
composer require crazy-max/cws-curl
```

And download the code:

```bash
composer install # or update
```

## Getting started

See `tests/test.php` file sample to help you.

## Example

![](.res/example.png)

## Methods

**reset** - Reset.<br />
**process** - Start the cURL request.<br />

**getUrl** - The URL to fetch.<br />
**setUrl** - Set the URL to fetch.<br />
**getMethod** - The HTTP request method.<br />
**setDeleteMethod** - Set DELETE HTTP request method.<br />
**setGetMethod** - Set GET HTTP request method. (default)<br />
**setHeadMethod** - Set HEAD HTTP request method.<br />
**setPostMethod** - Set POST HTTP request method.<br />
**setPutMethod** - Set PUT HTTP request method.<br />
**getParams** - Query string parameters.<br />
**addParam** - Add a custom parameter to the cURL request.<br />
**addOption** - Add an option for the cURL transfer.<br />
**getTimeout** - The maximum number of seconds to allow cURL functions to execute.<br />
**setTimeout** - Set the maximum number of seconds to allow cURL functions to execute.<br />
**getReferer** - The contents of the "Referer: " header to be used in a HTTP request.<br />
**setReferer** - Set the contents of the "Referer: " header to be used in a HTTP request.<br />
**getUserAgent** - The contents of the "User-Agent: " header to be used in a HTTP request.<br />
**setChromeUseragent** - Set the Chrome User-Agent to the contents of the "User-Agent: " header to be used in a HTTP request.<br />
**setFirefoxUseragent** - Set the Firefox User-Agent to the contents of the "User-Agent: " header to be used in a HTTP request.<br />
**setGooglebotUseragent** - Set the Googlebot User-Agent to the contents of the "User-Agent: " header to be used in a HTTP request.<br />
**setIeUseragent** - Set the Internet Explorer User-Agent to the contents of the "User-Agent: " header to be used in a HTTP request.<br />
**setOperaUseragent** - Set the Opera User-Agent to the contents of the "User-Agent: " header to be used in a HTTP request.<br />
**setUserAgent** - Set The contents of the "User-Agent: " header to be used in a HTTP request.<br />
**getUsername** - The username for the CURLOPT_USERPWD option.<br />
**getPassword** - The password associated to the username for the CURLOPT_USERPWD option.<br />
**setAuth** - Set authentication to the cURL request with username and password.<br />
**isRedirect** - Redirects allowed.<br />
**setRedirect** - Set allow redirects.<br />
**getMaxRedirect** - Maximum redirects allowed.<br />
**setMaxRedirect** - Set the maximum redirects allowed.<br />
**getProxyHost** - The host IP of the proxy to connect to.<br />
**getProxyPort** - The port number of the proxy to connect to.<br />
**getProxyType** - The proxy type CURLPROXY_HTTP, CURLPROXY_SOCKS4 or CURLPROXY_SOCKS5.<br />
**setProxy** - Set a HTTP proxy to tunnel requests through.<br />
**getProxyAuthType** - The HTTP authentication method(s) to use for the proxy connection. Can be CURLAUTH_BASIC or CURLAUTH_NTLM.<br />
**getProxyUsername** - The username for the CURLOPT_PROXYUSERPWD option.<br />
**getProxyPassword** - The password associated to the proxyUsername for the CURLOPT_PROXYUSERPWD option.<br />
**setProxyAuth** - Set a HTTP proxy authentication.<br /><br />

**getSession** - The current cURL session.<br />
**getStatus** - The HTTP status code returned.<br />
**getContent** - The content transferred.<br />
**getInfos** - The cURL information regarding the transfer.<br />
**getHeaderFulltext** - The header fulltext response.<br />
**getHeaders** - The headers response.<br />
**getError** - Get the last error.<br />

## How can I help ?

All kinds of contributions are welcome :raised_hands:!<br />
The most basic way to show your support is to star :star2: the project, or to raise issues :speech_balloon:<br />
But we're not gonna lie to each other, I'd rather you buy me a beer or two :beers:!

[![Paypal](.res/paypal.png)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=ECNZJ3UYAX7PU)

## License

MIT. See `LICENSE` for more details.


CHANGELOG.md
# Changelog

## 1.10.1 (2019/02/05)

* Switch to TravisCI (com)

## 1.10 (2018/01/04)

* Switch to MIT LICENSE

## 1.9 (2016/11/28)

* Add StyleCI config
* Fix short syntax for arrays
* Update dependencies

## 1.8 (2016/08/22)

* Scrutinizer issues

## 1.7 (2016/08/22)

* Add composer.json
* Add travis and scrutinizer
* Add CHANGELOG.md
* Review file structure
* Small refactoring

## 1.6 (2015/06/21)

* Adding reset method
* Bug setting default method

## 1.5 (2015/06/14)

* Refactoring
* Bug with referer

## 1.4 (2014/01/21)

* Adding LICENSE

## 1.3 (2013/05/20)

* Verbose quiet by default

## 1.2 (2013/05/19)

* Add follow location workaround
* Handling redirections with curl if safe_mode or open_basedir is enabled

## 1.1 (2013/05/17)

* Ability to add custom cURL options
* Adding proxy methods
* Spellchecking

## 1.0 (2013/05/17)

* Initial version


tests/test.php
<?php

require_once __DIR__.'/../vendor/autoload.php'; // Autoload files using Composer autoload

$cwsDebug = new Cws\CwsDebug();
$cwsDebug->setDebugVerbose();
$cwsDebug->setEchoMode();

// Start CwsCurl
$cwsCurl = new Cws\CwsCurl($cwsDebug);
$cwsCurl->setUrl('https://www.google.com'); // The URL to fetch
$cwsCurl->setGetMethod(); // HTTP request method ; default METHOD_GET
//$cwsCurl->setPostMethod();
//$cwsCurl->addParam("name", "value"); // Add custom parameters.
//$cwsCurl->addParam("name2", "value2");
//$cwsCurl->addOption(CURLOPT_SSL_VERIFYPEER, true); // Add an option for the cURL transfer.
//$cwsCurl->addOption(CURLOPT_ENCODING, "identity");
$cwsCurl->setTimeout(10); // The maximum number of seconds to allow cURL functions to execute ; default 10
//$cwsCurl->setReferer("http://www.example.com"); // The contents of the "Referer: " header
$cwsCurl->setFirefoxUserAgent(); // The contents of the "User-Agent: " header ; default UA_FIREFOX
//$cwsCurl->setUserAgent('Mozilla/5.0 (Windows NT 6.3; WOW64; rv:38.0) Gecko/20100101 Firefox/38.0');
//$cwsCurl->setAuth("admin", "admin"); // The username and password for the CURLOPT_USERPWD option
$cwsCurl->setRedirect(true); // Allow redirects ; default true
$cwsCurl->setMaxRedirect(3); // Maximum redirects allowed ; default 3
//$cwsCurl->setProxy("127.0.0.1", 1080, CURLPROXY_SOCKS5); // Set a HTTP proxy to tunnel requests through.
//$cwsCurl->setProxyAuth("admin", "admin", CURLAUTH_BASIC); // Set a HTTP proxy authentication.

// Process
$cwsCurl->process();


index.php
<?php

include('class.cws.curl.php');

$cwsCurl = new CwsCurl();
$cwsCurl->setDebugVerbose(CWSCURL_VERBOSE_DEBUG);               // default : CWSCURL_VERBOSE_QUIET
$cwsCurl->setUrl("http://www.google.com");                      // The URL to fetch
$cwsCurl->setMethod(CWSCURL_METHOD_GET);                        // HTTP request method ; default CWSCURL_METHOD_GET
//$cwsCurl->addParam("name", "value");                          // Add custom parameters.
//$cwsCurl->addParam("name2", "value2");
//$cwsCurl->addOption(CURLOPT_SSL_VERIFYPEER, true);            // Add an option for the cURL transfer.
//$cwsCurl->addOption(CURLOPT_ENCODING, "identity");
$cwsCurl->setTimeout(10);                                       // The maximum number of seconds to allow cURL functions to execute ; default 10
//$cwsCurl->setReferer("http://www.example.com");               // The contents of the "Referer: " header
$cwsCurl->setUserAgent(CWSCURL_UA_FIREFOX);                     // The contents of the "User-Agent: " header ; default CWSCURL_UA_FIREFOX
//$cwsCurl->setAuth("admin", "admin");                          // The username and password for the CURLOPT_USERPWD option
$cwsCurl->setRedirect(true);                                    // Allow redirects ; default true
$cwsCurl->setMaxRedirect(3);                                    // Maximum redirects allowed ; default 3
//$cwsCurl->setProxy("127.0.0.1", 1080, CURLPROXY_SOCKS5);      // Set a HTTP proxy to tunnel requests through.
//$cwsCurl->setProxyAuth("admin", "admin", CURLAUTH_BASIC);     // Set a HTTP proxy authentication.

// Process
$cwsCurl->process();

?>

example.php
<?php

// Download CwsDump at https://github.com/crazy-max/CwsDump
require_once '../CwsDump/class.cws.dump.php';
$cwsDump = new CwsDump();

// Download CwsDebug at https://github.com/crazy-max/CwsDebug
require_once '../CwsDebug/class.cws.debug.php';
$cwsDebug = new CwsDebug($cwsDump);
$cwsDebug->setDebugVerbose();
$cwsDebug->setEchoMode();

require_once 'class.cws.curl.php';
$cwsCurl = new CwsCurl($cwsDebug);

$cwsCurl->setUrl("https://www.google.com"); // The URL to fetch
$cwsCurl->setGetMethod(); // HTTP request method ; default METHOD_GET
//$cwsCurl->setPostMethod();
//$cwsCurl->addParam("name", "value"); // Add custom parameters.
//$cwsCurl->addParam("name2", "value2");
//$cwsCurl->addOption(CURLOPT_SSL_VERIFYPEER, true); // Add an option for the cURL transfer.
//$cwsCurl->addOption(CURLOPT_ENCODING, "identity");
$cwsCurl->setTimeout(10); // The maximum number of seconds to allow cURL functions to execute ; default 10
//$cwsCurl->setReferer("http://www.example.com"); // The contents of the "Referer: " header
$cwsCurl->setFirefoxUserAgent(); // The contents of the "User-Agent: " header ; default UA_FIREFOX
//$cwsCurl->setUserAgent('Mozilla/5.0 (Windows NT 6.3; WOW64; rv:38.0) Gecko/20100101 Firefox/38.0');
//$cwsCurl->setAuth("admin", "admin"); // The username and password for the CURLOPT_USERPWD option
$cwsCurl->setRedirect(true); // Allow redirects ; default true
$cwsCurl->setMaxRedirect(3); // Maximum redirects allowed ; default 3
//$cwsCurl->setProxy("127.0.0.1", 1080, CURLPROXY_SOCKS5); // Set a HTTP proxy to tunnel requests through.
//$cwsCurl->setProxyAuth("admin", "admin", CURLAUTH_BASIC); // Set a HTTP proxy authentication.

// Process
$cwsCurl->process();
